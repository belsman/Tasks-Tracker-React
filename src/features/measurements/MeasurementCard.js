import React from 'react';
import Slider from './MeasurementSlider';

const MeasurementCard = ({ record }) => {
  const { id, project, coding, reading, running, movie, created_at } = record;
  return (
    <section>
        <article className="record">
            <Slider title={created_at} recordId={id} />
            <h2>{created_at}</h2>
            <ul>
                <li>Project: {project}</li>
                <li>Coding: {coding}</li>
                <li>Reading: {reading}</li>
                <li>Running: {running}</li>
                <li>Movie: {movie}</li>
            </ul>
        </article>
  </section>
  );
};

export default MeasurementCard;
