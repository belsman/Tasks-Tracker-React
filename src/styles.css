* {
  box-sizing: border-box;
}

body {
  margin: 0;
  font-family: Arial, Helvetica, sans-serif;
  background-color: #e8e8e8;
  margin-bottom: 43px;
}

.h2 {
  font-size: large;
  font-weight: 700;
}

.navActive {
  background-color: #42b5e8;
}

.header {
  background-color: #42b5e8;
  color: #fff;
  padding: 1rem;
  text-align: center;
}
