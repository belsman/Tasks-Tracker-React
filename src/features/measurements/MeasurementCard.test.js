import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import pretty from 'pretty';
import { act } from 'react-dom/test-utils';

/* eslint-disable no-unused-vars */
import PropTypes from 'prop-types';
import { CircularProgressbar, buildStyles } from 'react-circular-progressbar';
import 'react-circular-progressbar/dist/styles.css';
import {
  RunningIcon, MovieIcon, CodingIcon, ReadingIcon, ProjectIcon,
} from '../../components/Icons';
import Slider from './MeasurementSlider';

import MeasurementCard from './MeasurementCard';

jest.mock('react-circular-progressbar', () => ({
  CircularProgressbar() {
    return (
      <div>A Dummy progressbar</div>
    );
  },
  buildStyles() {},
}));

jest.mock('../../components/Icons', () => ({
  RunningIcon() { return '<Icon />'; },
  MovieIcon() { return '<Icon />'; },
  CodingIcon() { return '<Icon />'; },
  ReadingIcon() { return '<Icon />'; },
  ProjectIcon() { return '<Icon />'; },
}));

jest.mock('./MeasurementSlider', () => function Slider() {
  return '<Slider />';
});

const record = {
  id: 1,
  running: 1,
  reading: 2,
  project: 3,
  coding: 4,
  movie: 5,
  created_at: '2013-03-10T02:00:00Z',
};

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
  act(() => {
    render(<MeasurementCard record={record} dailyTarget={20} />, container);
  });
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders the the progress pane', () => {
  const progressPane = document.querySelector('div.progressPane');
  expect(progressPane).not.toBeNull();
});

it('renders the correct number of tasks items', () => {
  const taskItems = document.querySelectorAll('li.taskItem');
  expect(taskItems).toHaveLength(5);
});

it('renders a Title for the project', () => {
  const projectTitle = document.querySelector('span.title');
  expect(projectTitle.textContent).toBe('Project');
});

it('renders a Value for the project', () => {
  const projectValue = document.querySelector('span.value');
  expect(projectValue.textContent).toBe('3h');
});

it('should render the expected output', () => {
  act(() => {
    render(<MeasurementCard record={record} dailyTarget={20} />, container);
  });

  expect(
    pretty(container.innerHTML),
  ).toMatchSnapshot();
});
