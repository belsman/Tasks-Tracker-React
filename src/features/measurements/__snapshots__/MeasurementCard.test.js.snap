// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the expected output 1`] = `
"<section>
  <article class=\\"record\\">&lt;Slider /&gt;<div class=\\"progressPane\\">
      <div class=\\"progressBarContainer\\">
        <div>A Dummy progressbar</div>
        <div class=\\"description\\">Total Time spent</div>
      </div>
      <div class=\\"progressBarContainer\\">
        <div>A Dummy progressbar</div>
        <div class=\\"description\\">Daily Targets</div>
      </div>
      <div class=\\"progressBarContainer\\">
        <div>A Dummy progressbar</div>
        <div class=\\"description\\">Progress</div>
      </div>
    </div>
    <ul class=\\"tasksData\\">
      <li class=\\"taskItem\\">&lt;Icon /&gt;<span><span class=\\"title\\">Project</span><span class=\\"value\\">3<span class=\\"unit\\">h</span></span></span></li>
      <li class=\\"taskItem\\">&lt;Icon /&gt;<span><span class=\\"title\\">Coding</span><span class=\\"value\\">4<span class=\\"unit\\">h</span></span></span></li>
      <li class=\\"taskItem\\">&lt;Icon /&gt;<span><span class=\\"title\\">Reading</span><span class=\\"value\\">2<span class=\\"unit\\">h</span></span></span></li>
      <li class=\\"taskItem\\">&lt;Icon /&gt;<span><span class=\\"title\\">Running</span><span class=\\"value\\">1<span class=\\"unit\\">h</span></span></span></li>
      <li class=\\"taskItem\\">&lt;Icon /&gt;<span><span class=\\"title\\">Movie</span><span class=\\"value\\">5<span class=\\"unit\\">h</span></span></span></li>
    </ul>
  </article>
</section>"
`;
